16S Microbiome Analysis in Qiita
================================



Analysis of Closed Reference processing
---------------------------------------

To create an analysis, select `Create new analysis` from the top menu. 

This will take you to a list of studies with samples available to you for
analysis, divided between your studies and publically available ('Other')
studies.

.. figure::  images/analysis-studies-page.png
   :align:   center

Find the study you created for this tutorial under "Your Studies". Click the
down arrow at the left of the row. This will expand the study to expose all
the objects from that study that are available to you for analysis.

.. figure::  images/analysis-study-expanded.png
   :align:   center

You could add all of these objects to the analysis by selecting the `Add to
Analysis` button. We will just add the Closed Reference OTU table object by
clicking `Add` in that row. 

.. figure::  images/analysis-closed-ref-add.png
   :align:   center

Now, the second-right-most icon at the top bar should be green, indicating that
there are samples selected for analysis. 

.. figure::  images/analysis-icon-green.png
   :align:   center

Clicking on the icon will take you to a page where you can refine the samples
you want to include in your analysis. Here, all 23 of our samples are currently
included:

.. figure::  images/analysis-samples-page.png
   :align:   center

You could optionally exclude particular samples from this set by clicking on
"Show/Hide samples", which will show each individual sample name along with a
"remove" option. (Removing them here will mask them from the analysis, but will
not affect the underlying files an any way.)

This should be good for now. Click the "Create Analysis" button, enter a name and
description, then click "Create analysis".

.. figure::  images/analysis-name-analysis.png
   :align:   center

This brings you to the analysis commands selection page, where you can specify
the steps in your analysis.

.. figure::  images/analysis-select-commands.png
   :align:   center

For this analysis, let's go ahead and select the commands Summarize
Taxa and Beta Diversity (Alpha Rarefaction can take some time to run). 

We will also need to specify an even sampling or rarefaction depth. All the
samples in the analysis will be randomly subsampled to this number of sequences,
reducing potential biases. Samples with fewer than this number of sequences
will be excluded, which can also be useful for excluding things like blanks.

You can get a good idea of where to set this threshold by looking at the
histogram generated by summarizing the input closed-reference OTU table, as
discussed in :doc:`qiita-16S-analysis`. Here, it looks like 2100 would be an
appropriate cutoff: it excludes one clear outlier, but retains most of the
samples.

.. figure::  images/analysis-closed-ref-histogram.png
   :align:   center

Enter 2100 in the rarefaction depth field, select the check boxes for
Summarize Taxa and Beta Diversity, and click "Start
Processing". You will see a list each step in the analysis, followed by its
status:

.. figure::  images/analysis-closed-ref-running.png
   :align:   center

When the analysis is finished, click the 'Success' link to see the results.

The results page will have sections indication which samples were dropped due
to insufficient numbers of reads, as well as sections for each data type. 

Here, we have taxonomy summaries and beta diversity PCoA plots available.

.. figure::  images/analysis-closed-ref-results.png
   :align:   center

Clicking on `bar_charts.html` under "Summarize Taxa" will take you to a
visualization of the taxa that were found in your sample:

.. figure::  images/analysis-closed-ref-barchart.png
   :align:   center

Under "Beta Diversity", you will have a selection of Principle Coordinates
Analyses of different measures of beta diversity, or the similarity between
samples.

Clicking on one (say, `unweighted unifrac emperor pcoa plot`) will open an
interactive visualization of the similarity among your samples. Generally
speaking, the more similar the samples, the closer the are likely to be in
the PCoA ordination. The Emperor visualization program offers a very useful way
to explore how patterns of similarity in your data associate with different
metadata categories. Here, I've colored the points in our test data by cat
ownership.

.. figure::  images/analysis-closed-ref-pcoa.png
   :align:   center

Let's take a few minutes now to explore the various features of Emperor. Open
a new browser window with the `Emperor tutorial <https://biocore.github.io/emperor/tutorial_index.html#section1>`__ and follow along with your test data.

Finally, if you ran Alpha Rarefaction, you will also have a link to interactive
plots that can be used to show how different measures of alpha diversity
correlate with different metadata categories:

.. figure::  images/analysis-closed-ref-alpha.png
   :align:   center


Analysis of deblur processing
-----------------------------

Creating an analysis of your deblurred data is virtually the same as the
process for the Closed Reference data, but there are a few quirks. 

First, because the deblur process creates two separate BIOM tables, you'll want
to make a note of the specific object ID number for the artifact you want to
use. In my case, that's ID 26017, the deblurred table with 'only-16s' reads.

.. figure::  images/analysis-deblur-object.png
   :align:   center

The specific ID for your table will be unique, so make a note of it, and you
can use it to select the correct table for analysis. 

Second, currently only the Beta Diversity analysis command option is working
with deblurred data. 


Creating a metaanalysis
----------------------

